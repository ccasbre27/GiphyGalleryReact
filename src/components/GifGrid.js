import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs( category );

    return (
        <div>
            <h3> { category } </h3>
            { loading && <p> Loading </p> }
            <div className="card-grid">
                <ol>
                    {
                        images.map( img => {
                            return <GifGridItem 
                                key={ img.id }
                                { ...img } />
                        })
                    }
                </ol>
            </div>
       </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
