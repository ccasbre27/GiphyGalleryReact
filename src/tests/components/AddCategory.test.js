import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Tests on <AddCategory />', () => {
   
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> ); 

    // se realiza antes de cada prueba
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> ); 
    });

    test('should render the component', () => {
    
        expect( wrapper ).toMatchSnapshot();
    
    });

    test('should change the input field', () => {
       
        const value = 'Hello World';
        const input = wrapper.find('input');
        
        input.simulate('change', { target: { value: value } });
       
    });
    
    test('should not post the information on submit event', () => {

        wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
       
        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('should call setCategories and clear the input field', () => {
        
        const value = 'Hello World';
        const input = wrapper.find('input');
       
        input.simulate('change', { target: { value: value } });
        wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
       
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) ); // we expect to had called this function sending a function as argument
        expect( input.prop('value') ).toBe('');

    });
    
    
    

});
