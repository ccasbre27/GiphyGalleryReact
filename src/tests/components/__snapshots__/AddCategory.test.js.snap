// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
